tools/src/wyvern/target/corewyvernIL/decl/Declaration.java

```diff
@@ -10,6 +10,16 @@
 import wyvern.tools.errors.FileLocation;
 
 public abstract class Declaration extends ASTNode implements EmitOIR {
+	private boolean hasResource = false;
+
+	public boolean containsResource() {
```

Comment:
Declare this abstract or with a default implementation, then override in subclasses. VarDeclaration always contains a resource; ValDeclaration contains one depending on it's type; TypeDeclaration never contains a resource.